### PR TITLE
Update of DPC dependencies for CI and conda-recipe

### PR DESCRIPTION
**Pull request by Alexsandruss**
## Description

Changes:
- Update DPC compiler, DPC runtime libraries and oneDAL development files to 2025 version in CI
- Update DPCTL and DPNP versions and their installation channel to PyPI for environment consistency
- Pin linux kernel headers in conda recipe to reflect GLIBC>=2.28 dependency of dal-devel dpc part
- Move DPC runtime libraries to optional dependencies in conda recipe

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

N/A, CI update.

### CODE REVIEW

File: .ci/pipeline/build-and-test-lnx.yml [hunk 1]
```diff
@@ -48,7 +45,7 @@ steps:
       bash .ci/scripts/setup_sklearn.sh $(SKLEARN_VERSION)
       pip install --upgrade -r requirements-test.txt
       pip install $(python .ci/scripts/get_compatible_scipy_version.py)
-      if [ $(echo $(PYTHON_VERSION) | grep '3.9\|3.11') ] && [ $(SKLEARN_VERSION) != "1.0" ]; then conda install -q -y -c https://software.repos.intel.com/python/conda/ dpctl=0.17.0 dpnp=0.15.0; fi
+      if [ $(SKLEARN_VERSION) != "1.0" ]; then pip install dpctl==0.18.* dpnp==0.16.*; fi
```
Comment (samir-nasibli) [inline .ci/pipeline/build-and-test-lnx.yml:48]: Are we installing dpctl/dpnp for all py versions?

Comment (Alexsandruss) [reply .ci/pipeline/build-and-test-lnx.yml:48]: Yes, it has same supported python versions as sklearnex.

Comment (icfaust) [reply .ci/pipeline/build-and-test-lnx.yml:48]: that does, however, cause certain functionality within sklearnex to become untested (see DummySyclQueue)

Comment (ethanglaser) [reply .ci/pipeline/build-and-test-lnx.yml:48]: Is DummySyclQueue needed if dpctl supports all sklearnex versions?

Comment (ethanglaser) [reply .ci/pipeline/build-and-test-lnx.yml:48]: Are there circumstances at this point where dpctl would not be available?

Comment (samir-nasibli) [reply .ci/pipeline/build-and-test-lnx.yml:48]: I can give at least two examples:
2. Installation Issues: problems on the user’s side during the installation of dpctl.
3. Environment Compatibility: for example, dpctl does not currently support Python 3.13. While scikit-learn already supports Python 3.13 and we may soon add it to our matrices, dpctl will not be available for this Python version at this time.

File: conda-recipe/meta.yaml [hunk 2]
```diff
@@ -58,29 +61,30 @@ requirements:
     - cython
     - jinja2
     - pybind11
-    - numpy {{ numpy }}
-    - impi-devel  # [not win]
+    - numpy
+    - mpich  # [not win]
```
Comment (ethanglaser) [inline conda-recipe/meta.yaml:65]: Why mpich instead of impi-devel?

**Review by ethanglaser — APPROVED** [review]
Thank you for resolving CI. preference would be to keep impi in meta.yaml if it makes sense.

File: conda-recipe/meta.yaml (hunk 2 above)
Comment (Alexsandruss) [reply conda-recipe/meta.yaml]: It was for testing purposes, mpi version doesn't make difference in public build of conda-recipe.

File: conda-recipe/meta.yaml [hunk 3]
```diff
@@ -58,15 +61,15 @@ requirements:
...
     - impi-devel  # [not win]
     # dal-devel pinning depends on the recipe location (repo or feedstock)
     - dal-devel
     # - dal-devel =={{ version }}
   run:
     - python
-    - {{ pin_compatible('numpy') }}
-    - dpcpp-cpp-rt  # [linux64]
+    - numpy
+    - scikit-learn
```
Comment (ethanglaser) [inline conda-recipe/meta.yaml:72]: Also, is scikit-learn needed in build deps or should it remain as test dep?

Comment (samir-nasibli) [reply conda-recipe/meta.yaml:72]: `scikit-learn` is only run time dependency.

**Review by icfaust — APPROVED** [review]
Before merge, @Alexsandruss could you leave a comment in the PR on the switch from conda to pip for dpctl/dpnp? This way when we face an upgrade issue with dpctl/dpnp in the future there is a bit of a note for someone to understand (doesn't need to be in the codebase, but something like research via git-blame to this PR would be nice)

**Comment by icfaust** [comment]
Also thank you for doing this, very much appreciated @Alexsandruss
